docs/source/how-to-guides/tutorial_symbol_and_string_collection.rst

```diff
@@ -0,0 +1,395 @@
+        "sha1": "38093fc0f043d0e639cc0b225e1acc038ffb7020",
+        "sha256": "83693b005ba387627ad10cef752d2559fe724cc0c7d4e86c4947f22403273e0c",
+        "sha512": null,
+        "git_sha1": null,
+        "mime_type": "text/x-c++",
+        "file_type": "C++ source, ASCII text",
+        "programming_language": "C",
+        "is_binary": false,
+        "is_text": true,
+        "is_archive": false,
+        "is_media": false,
+        "is_key_file": false,
+        "detected_license_expression": "",
+        "detected_license_expression_spdx": "",
+        "license_detections": [],
+        "license_clues": [],
+        "percentage_of_license_text": null,
+        "copyrights": [],
+        "holders": [],
+        "authors": [],
+        "package_data": [],
+        "emails": [],
+        "urls": [],
+        "extra_data": {
+            "source_symbols": [
+                "Bodies",
+                "BodyFarm",
+                "BodyFarm",
+                "BodyMap",
+                "C",
+                "Injector",
+                "LLVM_CLANG_LIB_ANALYSIS_BODYFARM_H",
+                "clang"
+            ]
+        }
+    }
+
+
+Xgettext Strings
+----------------
+
+- Send GET request to PurlDB with ``/api/collect/?purl=pkg:github/llvm/llvm-project@10.0.0
+  &addon_pipelines=collect_source_strings``.
+
+.. warning::
+    The ``collect_source_strings`` pipeline requires ``gettext``.
+
+- Once the indexing has completed visit ``/api/resources/?purl=pkg:github/llvm/llvm-project@10.0.0``
+  to get the ``source_strings`` for resources.
+
+- Below is the Xgettext strings for ``clang/include/clang/Analysis/BodyFarm.h``
+  file in ``extra_data`` field.
+
+.. code-block:: json
+
+    {
+        "package": "http://127.0.0.1:8001/api/packages/<package-id>/",
+        "purl": "pkg:github/llvm/llvm-project@10.0.0",
+        "path": "llvm-project-llvmorg-10.0.0.tar.gz-extract/llvm-project-llvmorg-10.0.0/clang/include/clang/Analysis/BodyFarm.h",
+        "type": "file",
+        "name": "BodyFarm.h",
+        "extension": ".h",
+        "size": 1509,
+        "md5": "808b7438da9841d95ae3a8135e7bf61f",
+        "sha1": "38093fc0f043d0e639cc0b225e1acc038ffb7020",
+        "sha256": "83693b005ba387627ad10cef752d2559fe724cc0c7d4e86c4947f22403273e0c",
+        "sha512": null,
+        "git_sha1": null,
+        "mime_type": "text/x-c++",
+        "file_type": "C++ source, ASCII text",
+        "programming_language": "C",
+        "is_binary": false,
+        "is_text": true,
+        "is_archive": false,
+        "is_media": false,
+        "is_key_file": false,
+        "detected_license_expression": "",
+        "detected_license_expression_spdx": "",
+        "license_detections": [],
+        "license_clues": [],
+        "percentage_of_license_text": null,
+        "copyrights": [],
+        "holders": [],
+        "authors": [],
+        "package_data": [],
+        "emails": [],
+        "urls": [],
+        "extra_data": {
```

Comment:
I think we should use an example that returns values in the `extra_data` field to give the reader an idea of what kind of values to expect.

Reply:
@JonoYang Fixed, now we're using a longer file for the tutorial, and we have return values for all pipelines.